fix: run team deletion on all hosts for replicated, non-sharded tables

Daesgar opened the pull request:
> [!IMPORTANT]
> 👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Problem

Replicated, non-sharded tables exist in the coordinator nodes as well.

Deletes are run there, but the delete dictionary does not exist, so the mutation never finishes and get stuck in a loop.

## Changes

Change the code so we create the deletion assets in all hosts of the cluster. When we run delete mutations for teams, wait for the mutation in all hosts.

In a separate PR, I will to set the `CLICKHOUSE_CLUSTER` environment variable to use the cluster that contains all hosts.

## Does this work well for both Cloud and self-hosted?

Yes.

## How did you test this code?

Tests.

Daesgar committed:
fix: run team deletion on one host for replicated, non-sharded tables

Daesgar committed:
fix: apply team deletes to coordinators as well

Daesgar committed:
Merge remote-tracking branch 'origin/master' into fix-team-deletes-coords

orian reviewed (approved):
Looks legit. Read all changes.

I'm wondering if there's a way we could test those dagster jobs?

Daesgar commented:
> Looks legit. Read all changes.
> 
> I'm wondering if there's a way we could test those dagster jobs?

We are running the Dagster tests that cover the end-to-end jobs. A while ago we introduced the coordinator in the docker-compose file to try and make the setup closer to our production setup, but it caused many trouble for everyone with connections / tests failing frequently so we removed it.
